app/controllers/restaurants_controller.rb

```diff
@@ -1,16 +1,62 @@
+require_relative './application_controller'
+require 'active_record'
+
 class RestaurantsController < ApplicationController
-  def list
+  before_action :set_restaurant, only: [:show, :update, :destroy]
+  #TODO: consider adding authentication
+  rescue_from ActiveRecord::RecordNotFound, :with => :invalid_id_handler
+
+  # GET /restaurants
+  def index
+    @restaurants = Restaurant.all
+    @restaurants = @restaurants.where('cuisine = ?', query_params["by_cuisine"]) if query_params.has_key? "by_cuisine"
+    @restaurants = @restaurants.where('rating >= ?', query_params["min_rating"].to_i) if query_params.has_key? "min_rating"
+    @restaurants = @restaurants.where('delivery_time <= ?', query_params["max_delivery_time"].to_i) if query_params.has_key? "max_delivery_time"
+    render :json => @restaurants
   end
 
-  def retrieve
+  # GET /restaurants/:id
+  def show
+    @restaurant
+    render :json => @restaurant
   end
 
+  #POST /restaurants
   def create
+    @restaurant = Restaurant.create(restaurant_params)
+    #TODO: could there be other errors other than bad request (failed model validations) from this object?
```

Comment:
To answer your question, when creating, usually there won't be, since you're not doing much. You could get BadRequest or ParameterMissing for example.
However, in all the controller there could be a mix of errors.
If you want to see how you can handle all of these potential errors in one place (using `concern`), instead of writing multiple `rescues` in each controller, I can show you what I did in my WeEat